third_party/terraform/utils/common_polling.go

```diff
@@ -0,0 +1,75 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+type PollWaiter interface {
+	CheckResponse(map[string]interface{}, error) (wait bool, err error)
+}
+
+func PollingWaitTime(config *Config, waiter PollWaiter, project, url, activity string, timeout time.Duration) error {
+	log.Printf("[DEBUG] %s: Polling until expected state is read", activity)
+
+	return resource.Retry(timeout, func() *resource.RetryError {
+		readResp, readErr := sendRequest(config, "GET", project, url, nil)
+		pollAgain, err := waiter.CheckResponse(readResp, readErr)
+		if pollAgain {
+			log.Printf("[DEBUG] Response has not terminated, continue polling: %v", err)
+			if err == nil {
+				err = errors.New("got pending response")
+			}
+			return resource.RetryableError(err)
+		}
+		log.Printf("[DEBUG] Response has terminated with error: %v", err)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+}
+
+// ResourceStatePoll implements PollWaiter. It should be used for resources where
+// a create/update operation returns the desired state and we are polling until the read state
+// converges. NOTE: This comparison to desired state is very dumb and must match exactly
+// for reflect.DeepEqual()
+// nolint: deadcode,unused
+type ResourceStatePollWaiter struct {
+	expected map[string]interface{}
+}
+
+// nolint: deadcode,unused
+func (p *ResourceStatePollWaiter) CheckResponse(res map[string]interface{}, err error) (bool, error) {
```

Comment:
Originally I wanted to use this, and it works for topic (which has one attribute, `name`) but subscription doesn't behave as well (the CREATE response returns an extra google-owned attribute that they remove for GET). I can take it out but it would allow for a more in-depth check against resource state.

Reply:
(and I filed a bug against P/S)